packages/coinjoin/src/client/round/outputDecomposition.ts

```diff
@@ -0,0 +1,212 @@
+/* eslint-disable no-await-in-loop */
+
+import * as coordinator from '../coordinator';
+import * as middleware from '../middleware';
+import { getRoundEvents, compareOutpoint, sumCredentials } from '../../utils/roundUtils';
+import { getExternalOutputSize } from '../../utils/coordinatorUtils';
+import type { Alice } from '../Alice';
+import type { CoinjoinRound, CoinjoinRoundOptions } from '../CoinjoinRound';
+
+/**
+ * RoundPhase: 2, step 1
+ *
+ * - Join all registered inputs Credentials in to one, grouped by accountKey
+ * - Calculate output amounts
+ */
+
+const joinInputsCredentials = async (
+    round: CoinjoinRound,
+    accountKey: string,
+    inputs: Alice[],
+    options: CoinjoinRoundOptions,
+) => {
+    const { roundParameters } = round;
+    const { signal, coordinatorUrl, middlewareUrl } = options;
+
+    let amountCredentials = inputs[0].confirmedAmountCredentials!;
+    let vsizeCredentials = inputs[0].confirmedVsizeCredentials!;
+    let { outputSize } = inputs[0];
+
+    options.log(`Joining Credentials for account <${accountKey}>. Total inputs: ${inputs.length}`);
+
+    if (inputs.length === 1) {
+        return {
+            accountKey,
+            amountCredentials,
+            vsizeCredentials,
+            outputSize,
+        };
+    }
+
+    for (let i = 1; i < inputs.length; i++) {
+        const current = inputs[i];
+        if (current.error) {
+            options.log(`Trying to join input with error. <${current.outpoint}> ${current.error}`);
+            throw current.error;
+        }
+        options.log(
```

Comment:
there's no need to redact those values in log?

Reply:
its only amounts, nothing sensitive